### PR TITLE
Streamlit webserver

### PR DESCRIPTION
**Pull request by RalfG**
Add code and Dockerfile for a minimal Streamlit webserver. User can upload data, set configuration options, and download zipped results. Basic plots showing the results are still missing, but should be feasible to implement in the near future. 

On each GitHub release, a docker image is built and pushed to ghcr.io.

### CODE REVIEW

File: streamlit/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+FROM python:3.8-buster
+ARG PERCOLATOR_VERSION=3-05
+COPY --from=percolator /tmp/percolator-noxml-v${PERCOLATOR_VERSION}-linux-amd64.deb /tmp
+COPY --from=percolator /tmp/percolator-converters-v${PERCOLATOR_VERSION}-linux-amd64.deb /tmp
+RUN dpkg -i /tmp/percolator-noxml-v${PERCOLATOR_VERSION}-linux-amd64.deb && \
+    dpkg -i /tmp/percolator-converters-v${PERCOLATOR_VERSION}-linux-amd64.deb && \
+    rm /tmp/percolator-noxml-v${PERCOLATOR_VERSION}-linux-amd64.deb && \
+    rm /tmp/percolator-converters-v${PERCOLATOR_VERSION}-linux-amd64.deb
+COPY . .
```
Comment (paretje) [inline streamlit/Dockerfile:16]: If you use a relative destination, you should declare `WORKDIR`

File: streamlit/ms2rescore_streamlit.py [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+        # Return results and cleanup files
+        st.markdown(
+            get_zipfile_href(
+                glob(config_dict["general"]["output_filename"] + "_*"),
+                filename="ms2rescore_results.zip",
+            ),
+            unsafe_allow_html=True,
+        )
+        shutil.rmtree(tmp_dir.name)
```
Comment (paretje) [inline streamlit/ms2rescore_streamlit.py:174]: You should use `tmp_dir.cleanup()` instead. Or use `tempfile.TemporaryDirectory` as a context manager.

File: streamlit/streamlit_utils.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    #MainMenu {visibility: hidden;}
+    </style>
+    """
+    st.markdown(hide_menu_style, unsafe_allow_html=True)
+
+
+def bytesio_to_tempfile(bytesio: BinaryIO) -> str:
+    """Write BytesIO object to temporary file."""
+    with tempfile.NamedTemporaryFile(delete=False) as tmp:
+        tmp.write(bytesio.getbuffer())
```
Comment (paretje) [inline streamlit/streamlit_utils.py:89]: It's probably better to use `shutil.copyfileobj`.

**Review by paretje — CHANGES_REQUESTED** [review]

**Comment by RalfG** [comment]
Closing in favor of local desktop GUI

**Pull request closed without merging**